test/modules/core/lib/pick-layers.spec.js

```diff
@@ -0,0 +1,450 @@
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+/* eslint-disable dot-notation, max-statements, no-unused-vars */
+
+import test from 'tape-catch';
+import {MapView, ScatterplotLayer, Deck, PolygonLayer, PathLayer} from 'deck.gl';
+import * as DATA from '../../../../examples/layer-browser/src/data-samples';
+
+const VIEW_STATE = {
+  latitude: 37.751537058389985,
+  longitude: -122.42694203247012,
+  zoom: 11.5,
+  pitch: 0,
+  bearing: 0
+};
+
+const TEST_CASES = [
+  {
+    id: 'scatterplotLayer',
+    props: {
+      width: 500,
+      height: 550,
+      layers: [
+        new ScatterplotLayer({
+          data: DATA.points,
+          getPosition: d => d.COORDINATES,
+          getColor: [255, 128, 0],
+          getRadius: d => d.SPACES,
+          opacity: 0.3,
+          pickable: true,
+          radiusScale: 30,
+          radiusMinPixels: 1,
+          radiusMaxPixels: 30
+        })
+      ],
+      views: [new MapView()],
+      viewState: VIEW_STATE,
+      useDevicePixels: false
+    },
+    pickingMethods: {
+      pickObject: [
+        {
+          parameters: {
+            x: 60,
+            y: 160
+          },
+          results: {
+            count: 1
+          }
+        },
+        {
+          parameters: {
+            x: 90,
+            y: 350
+          },
+          results: {
+            count: 0
+          }
+        }
+      ],
+      pickObjects: [
+        {
+          parameters: {
+            x: 300,
+            y: 300,
+            width: 100,
+            height: 100
+          },
+          results: {
+            count: 33
+          }
+        },
+        {
+          parameters: {
+            x: 50,
+            y: 50,
+            width: 10,
+            height: 10
+          },
+          results: {
+            count: 0
+          }
+        }
+      ],
+      pickMultipleObjects: [
+        {
+          parameters: {
+            x: 250,
+            y: 273
+          },
+          results: {
+            count: 2
+          }
+        },
+        {
+          parameters: {
+            x: 300,
+            y: 300
+          },
+          results: {
+            count: 0
+          }
+        }
+      ]
+    }
+  },
+  {
+    id: 'polygonLayer',
+    props: {
+      width: 500,
+      height: 550,
+      layers: [
+        new PolygonLayer({
```

Comment:
Many of these props are not necessary for the test. e.g. randomizing colors, elevation (extruded is not set). Trim the props that do not affect the rendering result.